src/cpu/translation.hh

```diff
@@ -254,7 +254,7 @@ class DataTranslation : public BaseMMU::Translation
            BaseMMU::Mode mode)
     {
         assert(state);
-        assert(mode == state->mode);
+        assert(mode == state->mode || req->isCacheClean());
```

Comment:
I feel a little uncomfortable with this change. However, if it is what makes the most sense (e.g., this can't be fixed by changing the original `Mode`, then I'm good with this change.

Thanks for the contribution!